pkg/noderesourcetopology/README.md

```diff
@@ -0,0 +1,224 @@
+# Overview
+
+This folder holds the Topology-aware scheduler plugin implementations based on [Topology awarescheduler plugin based on NodeResourceTopology CRD](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/kep/119-node-resource-topology-aware-scheduling/README.md). This plugin enables scheduling decisions based on worker node hardware topology overcoming the issue described [here](https://github.com/kubernetes/kubernetes/issues/84869).
+
+Document capturing the NodeResourceTopology API Custom Resource Definition Standard can be found [here](https://docs.google.com/document/d/12kj3fK8boNuPNqob6F_pPU9ZTaNEnPGaXEooW1Cilwg/edit).
+
+
+## Maturity Level
+
+<!-- Check one of the values: Sample, Alpha, Beta, GA -->
+
+- [ ] 💡 Sample (for demonstrating and inspiring purpose)
+- [x] 👶 Alpha (used in companies for pilot projects)
+- [ ] 👦 Beta (used in companies and developed actively)
+- [ ] 👨 Stable (used in companies for production workloads)
+
+## Tutorial
+
+### Expectation
+In case the cumulative count of node resource allocatable appear to be the same for both the nodes in the cluster, topology aware scheduler plugin uses the CRD instance corresponding to the nodes to obtain the resource topology information to make a topology-aware scheduling decision.
+
+### Config
+ filter must be enabled for topology-aware scheduling.
+```
+apiVersion: kubescheduler.config.k8s.io/v1beta1
+kind: KubeSchedulerConfiguration
+leaderElection:
+  leaderElect: false
+clientConnection:
+  kubeconfig: "/etc/kubernetes/scheduler.conf"
+profiles:
+- schedulerName: topo-aware-scheduler
+  plugins:
+    filter:
+      enabled:
+      - name: NodeResourceTopologyMatch
+# optional plugin configs
+  pluginConfig:
+  - name: NodeResourceTopologyMatch
+    args:
+      kubeconfigpath: "/etc/kubernetes/scheduler.conf"
+```
+
+### Demo
+
+Let us assume we have two nodes in a cluster deployed with sample-device-plugin with the hardware topology described by the diagram below:
+
+![Setup](numa-topology.png)
+
+The hardware topology corresponding to both the nodes is represented by the below CRD instances. These CRD instances are supposed to be created by Node Agents like [Resource Topology Exporter](https://github.com/k8stopologyawareschedwg/resource-topology-exporter) (RTE) or Node feature Discovery (NFD). Please refer to issue [Exposing Hardware Topology through CRDs in NFD](https://github.com/kubernetes-sigs/node-feature-discovery/issues/333) and [Design document](https://docs.google.com/document/d/1Q-4wSu1tzmbOXyGk_2r5_mK6JdXXJA-bOd3cAtBFnwo/edit?ts=5f24171f#) which captures details of enhancing NFD to expose node resource topology through CRDs.
+
+
+ ```yaml
+ # Master Node CRD spec
```

Comment:
it's not mandatory, here could be any node. Yes it might look strange since by default kubeadm installation  the master is not available for scheduling. And usually  on typical deployment master note is a VM, and definitely not a bare-metal.